### PR TITLE
Added adams-bashforth method of order 2, 3, 4, 5

### PR DESCRIPTION
**Pull request by ravi-ivar-7**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by tianyizheng02** [comment]
This PR also has the `runge_kitta_gills.py` file from your other PR, #10967, which has already been merged. Please remove this extraneous file.

File: maths/numerical_analysis/adams_bashforth.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+class AdamsBashforth:
+    def __init__(
+        self,
+        func: Callable[[float, float], float],
+        x_initials: list[float],
+        y_initials: list[float],
+        step_size: float,
+        x_final: float,
+    ):
```
Comment (cclauss) [inline maths/numerical_analysis/adams_bashforth.py:21]:
```suggestion
    ) -> None:
```
Or should this be a https://docs.python.org/3/library/dataclasses.html instead?

File: maths/numerical_analysis/adams_bashforth.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        if not all(x1 - x0 == step_size for x0, x1 in zip(x_initials, x_initials[1:])):
+            raise ValueError("x-values must be equally spaced according to step size.")
+
+        self.func = func
+        self.x_initials = x_initials
+        self.y_initials = y_initials
+        self.step_size = step_size
+        self.x_final = x_final
+
+    """
```
Comment (cclauss) [inline maths/numerical_analysis/adams_bashforth.py:39]: This needs to move up just after the function signature.

File: maths/numerical_analysis/adams_bashforth.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    y_initials: List containing initial required values of y.
+    step_size: The increment value of x.
+    x_final: The final value of x.
+
+    Returns: Solution of y at each nodal point
+
+    >>> def f(x, y):
+    ...     return x
+    >>> y = AdamsBashforth(f, [0, 0.2], [0, 0], 0.2, 1).step_2()
+    >>> y
```
Comment (cclauss) [inline maths/numerical_analysis/adams_bashforth.py:52]:
```suggestion
    >>> AdamsBashforth(f, [0, 0.2], [0, 0], 0.2, 1).step_2()
```

File: maths/numerical_analysis/adams_bashforth.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+    >>> def f(x,y):
+    ...     return x + y
+    >>> y = AdamsBashforth(f, [0, 0.2, 0.4, 0.6], [0, 0, 0.04, 0.128], 0.2, 1).step_4()
+    >>> y[-1]
+    0.57710833
+
+    >>> def f(x, y):
+    ...     return x + y
+    >>> y = AdamsBashforth(f, [0, 0.2, 1], [0, 0, 0.04], 0.2, 1).step_3()
```
Comment (cclauss) [inline maths/numerical_analysis/adams_bashforth.py:63]:
```suggestion
    >>> AdamsBashforth(f, [0, 0.2, 1], [0, 0, 0.04], 0.2, 1).step_3()
```

File: maths/numerical_analysis/adams_bashforth.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    Traceback (most recent call last):
+        ...
+    ValueError: The final value of x must be greater than the all initial values of x.
+
+    >>> def f(x, y):
+    ...     return x + y
+    >>> y = AdamsBashforth(f, [0, 0.2, 0.3], [0, 0, 0.04], 0.2, 1).step_3()
```
Comment (cclauss) [inline maths/numerical_analysis/adams_bashforth.py:70]:
```suggestion
    >>> AdamsBashforth(f, [0, 0.2, 0.3], [0, 0, 0.04], 0.2, 1).step_3()
```

File: maths/numerical_analysis/adams_bashforth.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    Traceback (most recent call last):
+        ...
+    ValueError: x-values must be equally spaced according to step size.
+
+    >>> def f(x, y):
+    ...     return x
+    >>> y = AdamsBashforth(f,[0,0.2,0.4,0.6,0.8],[0,0,0.04,0.128 0.307],-0.2,1).step_5()
```
Comment (cclauss) [inline maths/numerical_analysis/adams_bashforth.py:77]:
```suggestion
    >>> AdamsBashforth(f,[0,0.2,0.4,0.6,0.8],[0,0,0.04,0.128 0.307],-0.2,1).step_5()
```

File: maths/numerical_analysis/adams_bashforth.py [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    Traceback (most recent call last):
+        ...
+    ValueError: Step size must be positive.
+
+    >>> def f(x, y):
+    ...     return (x -y)/2
+    >>> y = AdamsBashforth(f, [0, 0.2, 0.4], [0, 0, 0.04], 0.2, 1).step_2()
```
Comment (cclauss) [inline maths/numerical_analysis/adams_bashforth.py:84]:
```suggestion
    >>> AdamsBashforth(f, [0, 0.2, 0.4], [0, 0, 0.04], 0.2, 1).step_2()
```

File: maths/numerical_analysis/adams_bashforth.py [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+    Traceback (most recent call last):
+        ...
+    ValueError: Insufficient nodal points values information.
+    """
+
+    def step_2(self) -> np.ndarray:
```
Comment (cclauss) [inline maths/numerical_analysis/adams_bashforth.py:90]: Please add doctests to each method.

**Review by cclauss — CHANGES_REQUESTED** [review]

**Review by cclauss — APPROVED** [review]
Let's do this!!

**Pull request merged**